### PR TITLE
Install django_extensions

### PR DESCRIPTION
**Pull request by chrismedrela**
`django_extensions` provide `shell_plus --notebook` command that open jupyter notebook in a browser, which is IMO much more convenient than command line shell.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,3 +21,5 @@ Requests-OAuthlib>=0.6.1,<0.7
 django-app-namespace-template-loader>=0.4
 django-webtest==1.8.0
 django-debug-toolbar==1.5
+django-extensions>=1.7,<1.8
+ipython[notebook]>=5.1,<5.2
```
Comment (pbanaszkiewicz) [inline requirements.txt]: Isn't this `jupyter[notebook]` nowadays?

**Comment by narayanaditya95** [comment]
@chrismedrela I usually use the IPython shell. Is there a specific reason to switch from the terminal to the brower?

**Comment by pbanaszkiewicz** [comment]
@narayanaditya95 for me the reason is mostly in `shell_plus` command itself, but `django_extensions` may contain some other nice things.

**Comment by chrismedrela** [comment]
> @chrismedrela I usually use the IPython shell. Is there a specific reason to switch from the terminal to the brower?

- Editing in browser is much more convenient than in console (especially if you have multiline snippet of code).
- Your session is saved in a notebook. A few days ago I created a notebook named issue-859 where I put experimental code needed to fix that issue. I love that today (when I came back to this issue) I could open this notebook and execute this code again.
- I can delete all unnecessary snippets of code. Or edit them in-place.
- If you want to restart shell, you just restart kernel and all your snippets of code are preserved.

**Comment by chrismedrela** [comment]
Additionally, django-extensions introduce `runserver_plus` that give you access to REPL in the case of an exception, so that you can debug in browser.

**Comment by pbanaszkiewicz** [comment]
My only concern is that we're forcing everyone to install jupyter even though it's strictly for Chris's development ease. We could add `dev_requirements.txt`.

**Comment by narayanaditya95** [comment]
>My only concern is that we're forcing everyone to install jupyter even though it's strictly for Chris's development ease. We could add dev_requirements.txt.

But then that would require manually adding `django_extensions` to the list of `INSTALLED_APPS`.

**Comment by chrismedrela** [comment]
> My only concern is that we're forcing everyone to install jupyter even though it's strictly for Chris's development ease. 

First of all, you can use these tools too. :-) But even if it's only for me, why is it a problem that we force everyone to install jupyter? It doesn't take a lot of time to install it.

**Comment by narayanaditya95** [comment]
@chrismedrela I think you can remove `jupyter` from `requirements.txt`. If someone is interested in using jupyter notebook, they can install it over the existing virtualenv.

**Pull request merged**